### PR TITLE
Adopt new hardware client as the default interface

### PR DESCRIPTION
**Pull request by exhaustin**
## Changelog

- [x] Test and debug hw_manual_test
- [x] Make `blocking=True` by default
- [x]  Remove old hw test script
- [x] Add deprecation warnings to other interfaces
- [x] Add examples
- [x] Update docs

TODO in future PR:
- [ ] Modify existing code to use new interface (put into stacked PR?)

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [x] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [x] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: examples/1_hw_basic_control.py [hunk 1]
```diff
@@ -17,8 +17,8 @@
     print(f"camera_pose={camera_pose}")
 
     # Move camera
-    robot.head.set_pan_tilt(pan=np.pi / 4, tilt=-np.pi / 3, blocking=True)
-    robot.head.look_at_ee(blocking=True)
+    robot.head.set_pan_tilt(pan=np.pi / 4, tilt=-np.pi / 3)
```
Comment (cpaxton) [inline examples/1_hw_basic_control.py:20]: i generally prefer using constants here, although this is an example so not sure

File: src/home_robot_hw/README.md [hunk 2]
```diff
@@ -45,7 +62,11 @@
...
 
-### Launching the Hello Stretch hardware drivers
+### On robot: Launching the Hello Stretch hardware drivers
 ```sh
 roslaunch home_robot startup_stretch_hector_slam.launch
-```
\ No newline at end of file
+```
+
+### On user machine: Interacting with the robot via the client
```
Comment (cpaxton) [inline src/home_robot_hw/README.md:70]: standardizing terms - "user machine" vs "server" or "workstation"?

File: src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py [hunk 3]
```diff
@@ -117,6 +117,10 @@ def __init__(
         Based on this code:
         https://github.com/hello-robot/stretch_ros/blob/master/hello_helpers/src/hello_helpers/hello_misc.py
         """
+        # Deprecation warning
+        rospy.logwarn(
+            "'StretchEnv' is being deprecated. Use 'home_robot_hw.StretchClient' to interact with the robot."
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py:122]: wait this whole env is being deprecated?

File: src/home_robot_hw/home_robot_hw/stretch_client/modules/head.py [hunk 4]
```diff
@@ -64,15 +64,15 @@ def set_pan_tilt(
...
         pan, tilt = self._robot_model.look_at_ee
         self.set_pan_tilt(pan, tilt, blocking=blocking)
 
-    def look_front(self, blocking: bool = False):
+    def look_front(self, blocking: bool = True):
         pan, tilt = self._robot_model.look_front
         self.set_pan_tilt(pan, tilt, blocking=blocking)
 
-    def look_ahead(self, blocking: bool = False):
+    def look_ahead(self, blocking: bool = True):
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/stretch_client/modules/head.py:75]: maybe comments to differntiate - look ahead i think is straight ahead, look front is angled down 45 degrees (navigation posture)

File: tests/hw_manual_test.py [hunk 5]
```diff
@@ -2,90 +2,53 @@
...
     # Manipulation
     print("Testing robot manipulation...")
     robot.switch_to_manipulation_mode()
 
-    pos_diff_goal = np.array([0.0, -0.2, 0.2])
-    pos, quat = model.get_ee_pose(q)
-    q = model.manip_ik((pos + pos_diff_goal, quat))
-    robot.goto(q, wait=True)
+    pos_diff_goal = np.array([0.2, -0.2, 0.2])
+    robot.manip.goto_ee_pose(pos_diff_goal, relative=True)
```
Comment (cpaxton) [inline tests/hw_manual_test.py:51]: why is this only 3? what happened to rotation? assumed to stay constant?

**Review by cpaxton — APPROVED** [review]
lgtm other than some minor comments

File: examples/1_hw_basic_control.py (hunk 1 above)
Comment (exhaustin) [reply examples/1_hw_basic_control.py:18]: Yeah seems like overkill for examples. I could define constants like DESIRED_TILT and stuff but I don't think it makes the code any more readable.

File: src/home_robot_hw/README.md (hunk 2 above)
Comment (exhaustin) [reply src/home_robot_hw/README.md:70]: I don't like the term "server" for the user machine as the robot is more of a server on the control side as it runs rosmaster (also I like the robotics as a service (RAAS) design methodology).
I'm using "robot" vs "user machine" for now but "workstation" also seems fine, not sure how ambiguous it is.
Would be nice to also get @zephirefaith 's input on this.

File: src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py (hunk 3 above)
Comment (exhaustin) [reply src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py:122]: That was my understanding, as all of the functionality this layer of abstraction provides is now inside `StretchClient`.
My original thought was that the hardware envs would directly inherit from `home_robot.Env`, and include `StretchClient` as a member variable (submodule), but open to discussion.
Maybe when I'm adapting the previous hardware envs to use the new interface, I can see if duplicated code end up existing across the hardware envs, then recreate `StretchEnv` to contain these duplications.

File: README.md [hunk 6]
```diff
@@ -11,13 +11,13 @@ Mostly Hello Stretch infrastructure
 
 ## Core Concepts
 
-This package assumes you have a low-cost mobile robot with limited compute -- initially a [Hello Robot Stretch](hello-robot.com/) - and a "server" workstation with more GPU compute. Both are assumed to be running on the same network.
+This package assumes you have a low-cost mobile robot with limited compute -- initially a [Hello Robot Stretch](hello-robot.com/) - and a "workstation" with more GPU compute. Both are assumed to be running on the same network.
```
Comment (zephirefaith) [inline README.md:14]: what do you mean "initially a Hello Robot Stretch"?

File: README.md [hunk 7]
```diff
@@ -154,9 +154,9 @@ conda activate contact_graspnet_env
 
 - `ImportError: cannot import name 'gcd' from 'fractions'`: Launch ros nodes from an env with Python 3.8 instead of 3.9
 
-### Syncing code between Robot and Server
+### Syncing code between Robot and Workstation
 
-Let `ROBOT_IP` store the robot's IP and let `SERVER_IP` store the server's IP. If your local network doesn't have access to internet we recommend using `rsync` with `--update` flag to sync your code changes across the machines. Usage:
+Let `ROBOT_IP` store the robot's IP and let `WORKSTATION_IP` store the workstation's IP. If your local network doesn't have access to internet we recommend using `rsync` with `--update` flag to sync your code changes across the machines. Usage:
 ```
 rsync -rvu /abs/path/to/local/home-robot $ROBOT_USER@$ROBOT_IP:/abs/path/to/remote/home-robot
```
Comment (zephirefaith) [inline README.md:161]: Could you add : "--filter=':- .gitignore'" at the end here?

File: examples/1_hw_basic_control.py [hunk 8]
```diff
@@ -1,9 +1,7 @@
-import time
```
Comment (zephirefaith) [inline examples/1_hw_basic_control.py:1]: Given this is an example, I would either add a readme with a one-liner about what this examples exhibits or a docstring at the top of script.

File: examples/2_hw_async_actions.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import time
```
Comment (zephirefaith) [inline examples/2_hw_async_actions.py:1]: Same comment as above about readme or module-level docstring

File: src/home_robot_hw/home_robot_hw/remote/modules/manip.py [hunk 10]
```diff
@@ -40,9 +40,20 @@ def _disable_hook(self) -> bool:
 
     # Interface methods
 
-    def get_ee_pose(self):
+    def get_ee_pose(self, world_frame=False):
```
Comment (zephirefaith) [inline src/home_robot_hw/home_robot_hw/remote/modules/manip.py:43]: type hints

File: src/home_robot_hw/home_robot_hw/remote/ros.py [hunk 11]
```diff
@@ -110,7 +110,7 @@ def __init__(
 
     def get_joint_state(self):
```
Comment (zephirefaith) [inline src/home_robot_hw/home_robot_hw/remote/ros.py:111]: return type hint

**Pull request merged**